### ISSUE TITLE
Scoping GraphQL queries to folders

### ISSUE THREAD

**Issue by felipesere** [gatsbyjs/gatsby#9104]
## Summary

I am creating a personal page with a couple of "sub-sections" that I have split into folders:
* /blogs/
* /portfolio/
* /photography/

When I added some markdown files under `/portfolio/` they spilled over into my blogs.
I was wondering if I now have to always "scope" the queries via the slug like so:
```
      filter: { fields: { slug: {re: "/portfolio" } } }
``` 

### Environment (if relevant)

  System:
    OS: macOS 10.14
    CPU: x64 Intel(R) Core(TM) i7-7700HQ CPU @ 2.80GHz
    Shell: 5.6.2 - /usr/local/bin/zsh
  Binaries:
    Node: 10.12.0 - /usr/local/bin/node
    npm: 6.4.1 - /usr/local/bin/npm
  Browsers:
    Chrome: 69.0.3497.100
    Firefox: 62.0.3
    Safari: 12.0
  npmPackages:
    gatsby: 2.0.17 => 2.0.17 
    gatsby-cli: ^2.4.3 => 2.4.3 
    gatsby-image: ^2.0.13 => 2.0.13 
    gatsby-plugin-manifest: ^2.0.4 => 2.0.4 
    gatsby-plugin-netlify: ^2.0.1 => 2.0.1 
    gatsby-plugin-offline: ^2.0.5 => 2.0.5 
    gatsby-plugin-react-helmet: ^3.0.0 => 3.0.0 
    gatsby-plugin-sass: ^2.0.1 => 2.0.1 
    gatsby-plugin-sharp: ^2.0.6 => 2.0.6 
    gatsby-plugin-typography: ^2.2.0 => 2.2.0 
    gatsby-remark-copy-linked-files: ^2.0.5 => 2.0.5 
    gatsby-remark-images: ^2.0.3 => 2.0.3 
    gatsby-remark-oembed: github:raae/gatsby-remark-oembed => 0.0.1 
    gatsby-remark-prismjs: ^3.0.1 => 3.0.1 
    gatsby-remark-responsive-iframe: ^2.0.5 => 2.0.5 
    gatsby-remark-smartypants: ^2.0.5 => 2.0.5 
    gatsby-source-filesystem: ^2.0.2 => 2.0.2 
    gatsby-transformer-remark: ^2.1.6 => 2.1.6 
    gatsby-transformer-sharp: ^2.1.3 => 2.1.3 
  npmGlobalPackages:
    gatsby-cli: 2.4.2

**Comment by pieh** [comment]
You are correct with your "guess". Just using `allMarkdownRemark` will get all markdown entries, so for blog / portfolio listings you will need to use some kind of filter. Using `slug` will work here just fine
